File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/PageInfo.java
package com.example.grts;

import viaduct.engine.api.EngineObjectData;
import viaduct.engine.api.NodeReference;
import viaduct.engine.api.RootFieldReference;
import viaduct.java.api.context.ExecutionContext;
import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.internal.InternalContext;
import viaduct.java.api.internal.NodeObjectBase;
import viaduct.java.api.internal.ObjectBase;
import viaduct.java.api.internal.OutputBuilderTypeChecker;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.RootObjectField;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.Arguments;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class PageInfo extends ObjectBase {

    public static final Type<PageInfo> Reflection = Type.ofClass(PageInfo.class);

    public static final class Fields implements TypeFields<PageInfo> {
        private Fields() {}

        public static final Field<PageInfo> __typename =
                Field.of("__typename", Reflection);
                public static final Field<PageInfo> hasNextPage =
                                Field.of("hasNextPage", Reflection);

                public static final Field<PageInfo> hasPreviousPage =
                                Field.of("hasPreviousPage", Reflection);

                public static final Field<PageInfo> startCursor =
                                Field.of("startCursor", Reflection);

                public static final Field<PageInfo> endCursor =
                                Field.of("endCursor", Reflection);

    }

    public PageInfo(InternalContext context, EngineObjectData.Sync data) {
        super(context, data);
    }

    @SuppressWarnings("UnusedMethod")
    private PageInfo(InternalContext context, Map<String, Object> data) {
        super(context, data, "PageInfo");
    }

    private PageInfo(InternalContext context, ObjectBase base, Map<String, Object> data) {
        super(context, base, data, "PageInfo");
    }

    public PageInfo(InternalContext context, RootFieldReference rootFieldReference) {
        super(context, rootFieldReference);
    }
        public boolean getHasNextPageOrThrow(String alias) {
            return fetchScalar("hasNextPage", alias);
        }

        public boolean getHasNextPageOrThrow() {
            return fetchScalar("hasNextPage", null);
        }

        public Boolean getHasNextPage(String alias) {
            return nullOnDataFailure(() -> fetchScalar("hasNextPage", alias));
        }

        public Boolean getHasNextPage() {
            return nullOnDataFailure(() -> fetchScalar("hasNextPage", null));
        }

        public boolean getHasPreviousPageOrThrow(String alias) {
            return fetchScalar("hasPreviousPage", alias);
        }

        public boolean getHasPreviousPageOrThrow() {
            return fetchScalar("hasPreviousPage", null);
        }

        public Boolean getHasPreviousPage(String alias) {
            return nullOnDataFailure(() -> fetchScalar("hasPreviousPage", alias));
        }

        public Boolean getHasPreviousPage() {
            return nullOnDataFailure(() -> fetchScalar("hasPreviousPage", null));
        }

        public String getStartCursorOrThrow(String alias) {
            return fetchScalar("startCursor", alias);
        }

        public String getStartCursorOrThrow() {
            return fetchScalar("startCursor", null);
        }

        public String getStartCursor(String alias) {
            return nullOnDataFailure(() -> fetchScalar("startCursor", alias));
        }

        public String getStartCursor() {
            return nullOnDataFailure(() -> fetchScalar("startCursor", null));
        }

        public String getEndCursorOrThrow(String alias) {
            return fetchScalar("endCursor", alias);
        }

        public String getEndCursorOrThrow() {
            return fetchScalar("endCursor", null);
        }

        public String getEndCursor(String alias) {
            return nullOnDataFailure(() -> fetchScalar("endCursor", alias));
        }

        public String getEndCursor() {
            return nullOnDataFailure(() -> fetchScalar("endCursor", null));
        }


    public Builder toBuilder() {
        return new Builder(__context(), toBuilderBase());
    }

    public static Builder builder(ExecutionContext context) {
        return new Builder(InternalContext.from(context), null);
    }

    public static class Builder {
        private final InternalContext __context;
        private final ObjectBase __base;
        private final Map<String, Object> data = new LinkedHashMap<>();

        private Builder(InternalContext __context, ObjectBase __base) {
            this.__context = __context;
            this.__base = __base;
        }

                public Builder hasNextPage(boolean hasNextPage) {
                    hasNextPage = OutputBuilderTypeChecker.checkField(
                            __context,
                            "PageInfo",
                            "hasNextPage",
                            null,
                            hasNextPage);
                    data.put("hasNextPage", hasNextPage);
        return this;
                }

                public Builder hasPreviousPage(boolean hasPreviousPage) {
                    hasPreviousPage = OutputBuilderTypeChecker.checkField(
                            __context,
                            "PageInfo",
                            "hasPreviousPage",
                            null,
                            hasPreviousPage);
                    data.put("hasPreviousPage", hasPreviousPage);
        return this;
                }

                public Builder startCursor(String startCursor) {
                    startCursor = OutputBuilderTypeChecker.checkField(
                            __context,
                            "PageInfo",
                            "startCursor",
                            null,
                            startCursor);
                    data.put("startCursor", startCursor);
        return this;
                }

                public Builder endCursor(String endCursor) {
                    endCursor = OutputBuilderTypeChecker.checkField(
                            __context,
                            "PageInfo",
                            "endCursor",
                            null,
                            endCursor);
                    data.put("endCursor", endCursor);
        return this;
                }


        public PageInfo build() {
            return new PageInfo(__context, __base, new LinkedHashMap<>(data));
        }
    }
}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/PageInfo.kt
@file:Suppress("warnings")

package viaduct.api.grts

import viaduct.apiannotations.InternalApi
import viaduct.api.context.ExecutionContext
import viaduct.api.internal.InternalContext
import viaduct.api.internal.ObjectBase
import viaduct.engine.api.EngineObject
import viaduct.engine.api.EngineObjectData

@OptIn(InternalApi::class)
class PageInfo(context: InternalContext, engineObject: EngineObject)
    : ObjectBase(context, engineObject), viaduct.api.types.Object
{
     fun getHasNextPageOrThrow(alias: String?): kotlin.Boolean = TODO()
     fun getHasNextPageOrThrow(): kotlin.Boolean = TODO()
     fun getHasNextPage(alias: String?): kotlin.Boolean? = TODO()
     fun getHasNextPage(): kotlin.Boolean? = TODO()

     fun getHasPreviousPageOrThrow(alias: String?): kotlin.Boolean = TODO()
     fun getHasPreviousPageOrThrow(): kotlin.Boolean = TODO()
     fun getHasPreviousPage(alias: String?): kotlin.Boolean? = TODO()
     fun getHasPreviousPage(): kotlin.Boolean? = TODO()

     fun getStartCursorOrThrow(alias: String?): kotlin.String? = TODO()
     fun getStartCursorOrThrow(): kotlin.String? = TODO()
     fun getStartCursor(alias: String?): kotlin.String? = TODO()
     fun getStartCursor(): kotlin.String? = TODO()

     fun getEndCursorOrThrow(alias: String?): kotlin.String? = TODO()
     fun getEndCursorOrThrow(): kotlin.String? = TODO()
     fun getEndCursor(alias: String?): kotlin.String? = TODO()
     fun getEndCursor(): kotlin.String? = TODO()


    fun toBuilder(): Builder =
        Builder(__context, __engineObject.type, toBuilderEOD())

    object of {
        operator fun invoke(context: ExecutionContext, block: Builder.() -> Unit): PageInfo =
            Builder(context).apply(block).build()
    }

    class Builder : ObjectBase.Builder<PageInfo> {
        constructor(context: ExecutionContext)
            : super(
                context as InternalContext,
                TODO() as graphql.schema.GraphQLObjectType,
                null
            )

        internal constructor(
            context: InternalContext,
            type: graphql.schema.GraphQLObjectType,
            baseEngineObjectData: EngineObjectData.Sync
        ) : super(context, type, baseEngineObjectData)

                  fun hasNextPage(value: kotlin.Boolean): Builder = TODO()

                  fun hasPreviousPage(value: kotlin.Boolean): Builder = TODO()

                  fun startCursor(value: kotlin.String?): Builder = TODO()

                  fun endCursor(value: kotlin.String?): Builder = TODO()


        final override fun build(): PageInfo = TODO()
    }

    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.PageInfo> {
        override final val name = "PageInfo"
        override final val kcls = viaduct.api.grts.PageInfo::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.PageInfo> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.PageInfo> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.PageInfo.Reflection)

            final val hasNextPage: viaduct.api.reflect.Field<viaduct.api.grts.PageInfo> =
                viaduct.api.internal.FieldImpl("hasNextPage", viaduct.api.grts.PageInfo.Reflection)

            final val hasPreviousPage: viaduct.api.reflect.Field<viaduct.api.grts.PageInfo> =
                viaduct.api.internal.FieldImpl("hasPreviousPage", viaduct.api.grts.PageInfo.Reflection)

            final val startCursor: viaduct.api.reflect.Field<viaduct.api.grts.PageInfo> =
                viaduct.api.internal.FieldImpl("startCursor", viaduct.api.grts.PageInfo.Reflection)

            final val endCursor: viaduct.api.reflect.Field<viaduct.api.grts.PageInfo> =
                viaduct.api.internal.FieldImpl("endCursor", viaduct.api.grts.PageInfo.Reflection)

    }

}